piksi_tools/console/settings_view.py

```diff
@@ -222,33 +182,43 @@ def revert_to_prior_value(self, name, old, new, error_value=SBP_WRITE_STATUS.TIM
+            if type(self.value) == unicode:
+                self.value = self.value.encode('ascii', 'replace')
+            self.confirmed_set = False
+            res = self.settings.s_api.write(self.section, self.name, new)
+            if res == settings_write_res_e.SETTINGS_WR_OK:
+                self.value = new
+            else:
+                self.revert_to_prior_value(self.name, old, new, res)
+
+            self.confirmed_set = True
+
+        self.write_event.set()
+
+        # If we have toggled the Inertial Nav enable setting (currently "output mode")
+        # we display some helpful hints for the user
+        if (self.section == "ins" and self.name == "output_mode" and
+                old is not None and self.settings is not None):
+            if new in ['GNSS and INS', 'INS Only', 'Loosely Coupled', 'LC + GNSS', 'Debug', 'debug']:
+                hint_thread = threading.Thread(
+                    target=self.settings._display_ins_settings_hint)
+                hint_thread.start()
+            # regardless of which way setting is going, a restart is required
+            else:
+                self.settings.display_ins_output_hint()
 
     def _value_changed(self, name, old, new):
-        '''When a user changes a value, kick off a timed revert thread to revert it in GUI if no confirmation
-            that the change was successful is received.'''
-        if not self._prevent_revert_thread:
-            if getattr(self, 'settings', None):
-                if (old != new and old is not Undefined and new is not Undefined):
-                    if type(self.value) == unicode:
-                        self.value = self.value.encode('ascii', 'replace')
-                    self.confirmed_set = False
-                    self.timed_revert_thread = TimedDelayStoppableThread(
-                        SETTINGS_REVERT_TIMEOUT,
-                        target=self.revert_to_prior_value,
-                        args=(name, old, new))
-                    self.settings.set(self.section, self.name, self.value)
-                    self.timed_revert_thread.start()
-                # If we have toggled the Inertial Nav enable setting (currently "output mode")
-                # we display some helpful hints for the user
-                if (self.section == "ins" and self.name == "output_mode" and
-                        old is not None and self.settings is not None):
-                    if new in ['GNSS and INS', 'INS Only', 'Loosely Coupled', 'LC + GNSS', 'Debug', 'debug']:
-                        hint_thread = threading.Thread(
-                            target=self.settings._display_ins_settings_hint)
-                        hint_thread.start()
-                    # regardless of which way setting is going, a restart is required
-                    else:
-                        self.settings.display_ins_output_hint()
+        if getattr(self, 'settings', None) is None:
+            return
+
+        if self.reverting:
```

Comment:
I think it also makes sense to avoid firing a thread (or a task) if `old != new`.

Reply:
Rather `old == new` you mean?

Reply:
Yes